security/security-api/custom-identity-store/README.md

```diff
@@ -0,0 +1,175 @@
+# Custom Identity Store
+
+An application also can provide its own IdentityStore. Bundled with the application, this is the identity store, which is then used for authentication, authorization.
+
+This example demonstrates how a custom identity store `TestIdentityStore` can be defined and provided as part of the application being deployed.
+
+In this example, following user is defined along with the roles they are in.
+
+|User|Password|Role|
+|----|--------|----|
+|reza|secret1|foo,bar|
+
+When a request is made to the application with certain credentials, the authentication mechanism bundled with this application comes into affect and an authentication is done against the `TestIdentityStore` as defined in the application.
+
+Post authentication, the application will also verify the roles the caller is in and will send the details as part of the response.
+
+Defining credentials and the roles assigned to the users is done as shown below in `TestIdentityStore`:
+
+```java
+if (usernamePasswordCredential.compareTo("reza", "secret1")) {
+    return new CredentialValidationResult("reza", new HashSet<>(asList("foo", "bar")));
+}
+
+```
+
+With application, a custom `HttpAuthenticationMechanism` too is bundled which gets called during authentication process.
```

Comment:
A custom 'HttpAuthenticationMechanism' is bundled with the application and gets called during the authentication process.